packages/wrangler/src/pages/deploy.tsx

```diff
@@ -93,6 +93,17 @@ export const Handler = async (args: PagesDeployArgs) => {
 		);
 	}
 
+	if (args.experimentalJsonConfig) {
+		throw new FatalError("Pages does not support `wrangler.json`", 1);
+	}
+
+	if (args.env) {
+		throw new FatalError(
+			"Pages does not support imperatively targeting a particular environment",
```

Comment:
not sure if `imperatively targeting a particular environment` is the best wording here. Happy for suggestions

Reply:
Maybe something like: `Pages does not support targeting an environment with the --env flag. Use the --branch flag to target your production or preview branch`?

Reply:
Yeah @penalosa's wording sounds good. Makes it clear that it is the --env flag that is at fault here.